File: WordPig/app/src/androidTest/java/com/pigdogbay/wordpig/model/WordCheckerTest.java
package com.pigdogbay.wordpig.model;

import android.test.AndroidTestCase;

import com.pigdogbay.library.diagnostics.Timing;
import com.pigdogbay.library.utils.LineReader;
import com.pigdogbay.wordpig.R;


import java.util.List;

public class WordCheckerTest extends AndroidTestCase {

    public void testIsWord1() throws Exception {
        List<String> wordList = LineReader.Read(getContext(), R.raw.standard);
        WordChecker target = new WordChecker(wordList);
        assertTrue(target.isWord("test"));
        assertFalse(target.isWord("qwsdt"));
    }

    public void testLoadTimes() throws Exception {
        List<String> wordList = null;

        Timing timing = new Timing();
        for (int i=0;i<10;i++) {
            wordList = LineReader.Read(getContext(), R.raw.standard);
        }
        timing.LogDuration("tests");
    }

    public void testSearchTimes() throws Exception {
        List<String> wordList = LineReader.Read(getContext(), R.raw.standard);
        WordChecker target = new WordChecker(wordList);

        Timing timing = new Timing();
        for (int i=0;i<10;i++) {
            assertTrue(target.isWord("test"));
        }
        timing.LogDuration("tests");
    }

}

File: WordPig/app/src/androidTest/java/com/pigdogbay/wordpig/model/WordCheckerTest.kt
package com.pigdogbay.wordpig.model

import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.platform.app.InstrumentationRegistry
import com.pigdogbay.lib.diagnostics.Timing
import com.pigdogbay.lib.utils.LineReader
import com.pigdogbay.wordpig.R
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class WordCheckerTest {
    @Test
    fun testIsWord1() {
        val wordList: List<String> = LineReader.Read(InstrumentationRegistry.getInstrumentation().targetContext, R.raw.standard)
        val target = WordChecker(wordList)
        assertTrue(target.isWord("test"))
        assertFalse(target.isWord("qwsdt"))
    }

    @Test
    fun testLoadTimes() {
        var wordList: List<String?>? = null
        val timing = Timing()
        for (i in 0..9) {
            wordList = LineReader.Read(InstrumentationRegistry.getInstrumentation().targetContext, R.raw.standard)
        }
        timing.logDuration("tests")
    }

    @Test
    fun testSearchTimes() {
        val wordList: List<String> = LineReader.Read(InstrumentationRegistry.getInstrumentation().targetContext, R.raw.standard)
        val target = WordChecker(wordList)
        val timing = Timing()
        for (i in 0..9) {
            assertTrue(target.isWord("test"))
        }
        timing.logDuration("tests")
    }
}
